sky/task.py

```diff
@@ -581,94 +724,10 @@ def sync_storage_mounts(self) -> None:
-            run on, and local is the node from which the task is launched.
-        """
-        if file_mounts is None:
-            self.file_mounts = None
-            return self
-        for target, source in file_mounts.items():
-            if target.endswith('/') or source.endswith('/'):
-                with ux_utils.print_exception_no_traceback():
-                    raise ValueError(
-                        'File mount paths cannot end with a slash '
-                        '(try "/mydir: /mydir" or "/myfile: /myfile"). '
-                        f'Found: target={target} source={source}')
-            if data_utils.is_cloud_store_url(target):
-                with ux_utils.print_exception_no_traceback():
-                    raise ValueError(
-                        'File mount destination paths cannot be cloud storage')
-            if not data_utils.is_cloud_store_url(source):
-                if not os.path.exists(
-                        os.path.abspath(os.path.expanduser(source))):
-                    with ux_utils.print_exception_no_traceback():
-                        raise ValueError(
-                            f'File mount source {source!r} does not exist '
-                            'locally. To fix: check if it exists, and correct '
-                            'the path.')
-            # TODO(zhwu): /home/username/sky_workdir as the target path need
-            # to be filtered out as well.
-            if (target == constants.SKY_REMOTE_WORKDIR and
-                    self.workdir is not None):
-                with ux_utils.print_exception_no_traceback():
-                    raise ValueError(
-                        f'Cannot use {constants.SKY_REMOTE_WORKDIR!r} as a '
-                        'destination path of a file mount, as it will be used '
-                        'by the workdir. If uploading a file/folder to the '
-                        'workdir is needed, please specify the full path to '
-                        'the file/folder.')
-
-        self.file_mounts = file_mounts
-        return self
-
-    def update_file_mounts(self, file_mounts: Dict[str, str]):
-        """Updates the file mounts for this Task.
-
-        This should be run before provisioning.
-
-        Example:
-
-            task.update_file_mounts({
-                '~/.config': '~/Documents/config',
-                '/tmp/workdir': '/local/workdir/cnn-cifar10',
-            })
-
-        Args:
-          file_mounts: a dict of { remote_path: local_path }, where remote is
-            the VM on which this Task will eventually run on, and local is the
-            node from which the task is launched.
-        """
-        if self.file_mounts is None:
-            self.file_mounts = {}
-        self.file_mounts.update(file_mounts)
-        # For validation logic:
-        return self.set_file_mounts(self.file_mounts)
-
     def get_local_to_remote_file_mounts(self) -> Optional[Dict[str, str]]:
```

Comment:
Do we need to expose this? Also, for `get_cloud_to_remote_file_mounts` and `to_yaml_config`

Reply:
Good call, agreed it's a good idea to minimize the advertised API surface. All hidden now.